api/app/helpers/spree/api/api_helpers.rb

```diff
@@ -37,7 +37,16 @@ module ApiHelpers
         :variant_property_attributes
       ]
 
-      mattr_reader(*ATTRIBUTES)
+      ATTRIBUTES.each do |attribute|
+        define_method attribute do
+          Spree::Api::Config.send(attribute)
+        end
+
+        define_singleton_method attribute do
+          Spree::Deprecation.warn("Please use Spree::Api::Config::#{attribute} instead.")
+          Spree::Api::Config.send(attribute)
+        end
```

Comment:
I'm not sure to get why we need both `define_method` and `define_singleton_method` here. Can you please help me with some details?

Reply:
Sure. Methods defined with `define_method` are the ones used in views. Remove that definition and you'll see every test rendering something using these methods fail.

The singleton methods are the ones called when you reference to the helper directly, like so:

```
2.7.0 :015 > Spree::Api::ApiHelpers.user_attributes
DEPRECATION WARNING: Please use Spree::Api::Config::user_attributes instead. (called from irb_binding at (irb):15)
 => [:id, :email, :created_at, :updated_at, :likes_count]
```

You can clearly see the deprecation warning. Without that definition every piece of code currently using will break (like some of our tests, that I changed in this PR). With this code you will get the deprecation warning.

Plus, the very definition of [`mattr_reader`](https://apidock.com/rails/Module/mattr_reader) (previously used to generate these methods) explains how it creates both a class and an instance reader for the provided symbol.

I just pushed a commit with tests regarding the deprecation warning.